added v1beta1 version for PodPreset APIs

droot opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://git.k8s.io/community/contributors/devel/pull-requests.md#the-pr-submit-process and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/devel/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/devel/pull-requests.md#write-release-notes-if-needed
-->

**What this PR does / why we need it**:
Adds v1beta1 version for PodPresets API

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
v1beta1 version starts with v1alpha1 features and we will add following enhancements to v1beta1:
 - container targeting support
 - improve the visibility in case of conflicts
 - extend information injection to Initcontainers

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```release-note
PodPresets v1beta1 version
```

droot commented:
cc @caesarxuchao @pwittrock

caesarxuchao commented on cmd/kube-apiserver/app/aggregator.go:
```diff
@@ -188,6 +188,7 @@ var apiVersionPriorities = map[schema.GroupVersion]priority{
 	{Group: "rbac.authorization.k8s.io", Version: "v1beta1"}:     {group: 17000, version: 12},
 	{Group: "rbac.authorization.k8s.io", Version: "v1alpha1"}:    {group: 17000, version: 9},
 	{Group: "settings.k8s.io", Version: "v1alpha1"}:              {group: 16900, version: 9},
+	{Group: "settings.k8s.io", Version: "v1beta1"}:               {group: 16900, version: 9},
```
I think you should give v1beta1 a higher priority.

droot replied:
We are not ready to make v1beta1 higher priority because this change just introduces v1beta1 version so that we can add a few enhancements to it. We will make v1beta1 higher priority when we are done with all the v1beta1 enhancements and it is sufficiently tested.

caesarxuchao replied:
I don't know if make them a tie will cause undefined behavior.

pmorie replied:
1. we should find out what the behavior of matching priorities for different versions of the same group is
2. how about make v1beta1 lower priority for now?

droot replied:
1: v1alpha1 is given higher priority over v1beta1 in case of a tie.
2: made v1beta1 lower priority than v1alpha1 for now.

caesarxuchao commented:
One comment, otherwise the apimachinery and generated code looks good.

Could you move those generated files from the first commit to the second commit? This PR could serve as an example for other contributors. Thanks.

droot commented:
@caesarxuchao got rid of generated files from the first commit. So hopefully, it can serve as a reference for adding a new version. PTAL.

droot commented:
/assign @smarterclayton

pmorie reviewed (commented):
Seems pretty straight forward - are there any kubectl changes required?

pmorie commented:
@k8s-bot ok to test

droot commented:
@caesarxuchao @pmorie addressed the comments and updated the PR. PTAL. 

@pmorie no changes were required in kubectl side. I did a manual test where we v1beta1 API is compatible with v1alpha1.

droot commented:
cc @lavalamp

droot committed:
add v1beta1 version for PodPreset APIs

v1beta1 is disabled by default and will be enabled
by default after all beta enhancements are added (1.8)

droot committed:
added generated files for v1beta1 for PodPresets

droot commented:
/retest

droot commented:
/assign @erictune

cblecker commented:
cross-build is currently broken: https://github.com/kubernetes/kubernetes/issues/48887

cblecker commented:
/test pull-kubernetes-cross

droot commented:
It is decided that PodPresets will be moved to service catalog repo. So Beta version of this API will be added in service catalog repo. So abandoning this PR.
